src/commands/variable-group/create.ts

```diff
@@ -1,96 +1,78 @@
 import { logger } from "../../logger";
 import { IVariableGroupData } from "../../types";
+import decorator from "./create.decorator.json";
 
 /**
  * Adds the create command to the variable-group command object
  *
  * @param command Commander command object to decorate
  */
-export const createCommandDecorator = (command: commander.Command): void => {
-  command
-    .command("create")
-    .alias("c")
-    .description("Add a new variable group in Azure DevOps project.")
-    .option(
-      "-f, --file <variable-group-manifest-file-path>",
-      "Path to the yaml file that contains variable group manifest."
-    )
-    .option(
-      "-o, --org-name <organization-name>",
-      "Azure DevOps organization name; falls back to azure_devops.org in spk config"
-    )
-    .option(
-      "-p, --project <project>",
-      "Azure DevOps project name; falls back to azure_devops.project in spk config"
-    )
-    .option(
-      "-t, --personal-access-token <personal-access-token>",
-      "Personal access token associated with the Azure DevOps org; falls back to azure_devops.access_token in spk config"
-    )
-    .action(async opts => {
-      try {
-        if (!opts.file) {
-          logger.error(
-            "You need to specify a file with variable group manifest"
-          );
-          return;
-        }
-
-        const { file, orgName, project, personalAccessToken } = opts;
-
-        logger.debug(
-          `opts: ${file}, ${orgName}, ${project}, ${personalAccessToken}`
+export const commandDecorator = (command: commander.Command): void => {
+  buildCmd(command, decorator).action(async opts => {
+    try {
+      if (!opts.file) {
+        throw new Error(
+          "You need to specify a file with variable group manifest"
         );
+      }
 
-        // type check
-        if (typeof orgName !== "undefined" && typeof orgName !== "string") {
-          throw Error(
-            `--org-name must be of type 'string', ${typeof orgName} specified.`
-          );
-        }
-
-        if (typeof project !== "undefined" && typeof project !== "string") {
-          throw Error(
-            `--project must be of type 'string', ${typeof project} specified.`
-          );
-        }
+      const { file, orgName, project, personalAccessToken } = opts;
 
-        if (
-          typeof personalAccessToken !== "undefined" &&
-          typeof personalAccessToken !== "string"
-        ) {
-          throw Error(
-            `--personal-access-token must be of type 'string', ${typeof personalAccessToken} specified.`
-          );
-        }
+      logger.debug(
+        `opts: ${file}, ${orgName}, ${project}, ${personalAccessToken}`
+      );
 
-        const accessOpts: IAzureDevOpsOpts = {
-          orgName,
-          personalAccessToken,
-          project
-        };
-        logger.debug(`access options: ${JSON.stringify(accessOpts)}`);
+      // type check
+      if (typeof orgName !== "undefined" && typeof orgName !== "string") {
```

Comment:
Are these type checks still valid with command builder implementation?

Reply:
I will not make these changes because I think we are going to combine this one command with another one. :-)